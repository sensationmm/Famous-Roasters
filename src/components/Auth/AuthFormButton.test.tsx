import { MockedProvider } from '@apollo/client/testing'
import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { AuthFormButton } from '.'

describe('Auth form button custom renderer on auth component', () => {
  const snippet = () => (
    <form name="testAuthFormButton">
      <AuthFormButton ctaText="ctaText" />
    </form>
  )

  it('Renders correctly', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>{snippet()}</MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
