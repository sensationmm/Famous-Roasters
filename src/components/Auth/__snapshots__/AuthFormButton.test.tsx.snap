// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth form button custom renderer on auth component Renders correctly 1`] = `
<div>
  <form
    name="testAuthFormButton"
  >
    <button
      class="flex w-full justify-center inline-flex items-center rounded-full border-2 px-4 py-2 text-md font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
      data-testid="submit"
      role="button"
      type="submit"
    >
      ctaText
    </button>
  </form>
</div>
`;
